scipy/special/_basic.py

```diff
@@ -2953,6 +2956,108 @@ def factorialk(n, k, exact=True):
         raise NotImplementedError
 
 
+def stirling2(N, K, exact=True):
+    """Generate Stirling number(s) of the second kind.
+
+        Stirling numbers of the second kind count the number
+        of non-empty subsets that can be formed by N elements
+        into K non-empty subsets.
+
+        This is often expressed as "N subset K".
+
+        Parameters
+        ----------
+        N : int, ndarray
+            Number of things.
+        K : int, ndarray
+            Number of non-empty subsets taken.
+        exact : bool, optional
+            For integers, if `exact` is False, then floating point precision is
+            used, otherwise the result is computed exactly. For non-integers,
+            if `exact` is True, is truncated. Defaults to True.
+
+        Returns
+        -------
+        val : int, float, ndarray
+            The number of partitions.
+
+        See Also
+        --------
+        comb : The number of combinations of N things taken k at a time.
+
+        Notes
+        -----
+        - If N < 0, or K < 0, then 0 is returned.
+        - If K > N, then 0 is returned.
+        The output type of the ndarray is always `object`.
+        The input type of the ndarrays must both be integer otherwise a
+        TypeError is raised.
+
```

Comment:
You should add a note advertising that when array like input is passed, this implementation avoids duplicating effort across computing the different values. It's an awesome feature which needs to be pointed out.